### PR TITLE
Clearer and more exhaustive string benchmark

### PR DESCRIPTION
**Pull request by Wicpar**
- updates crate versions for bench
- enabled html reports
- grouped benches in better way for criterion
- made benches clearer and more reusable.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1301,6 +1301,12 @@ mod cache_string {
             Self(array)
         }
     }
+
+    impl From<&str> for CacheString {
```
Comment (paulocsanz) [inline src/lib.rs:1305]: Nice catch!

**Review by paulocsanz — APPROVED** [review]
Awesome changes, thank you!

**Comment by Wicpar** [comment]
I'll do the same some time this week with the unit tests using [The test-case crate](https://github.com/frondeus/test-case/wiki/Syntax) 

It looks like one of the most fleshed out creates to do bulk tests.

**Comment by paulocsanz** [comment]
That's super cool! Seems to also lay the ground for continuous fuzzing.

**Pull request merged**